.github/actions/setup/action.yml

```diff
@@ -1,10 +1,22 @@
 name: Setup
-description: Configure the common enviroment
+description: Common environment setup
 
 runs:
   using: composite
   steps:
+    - uses: actions/setup-node@v4
+      with:
+        node-version: 20.x
+    - uses: actions/cache@v4
+      id: cache
+      with:
+        path: '**/node_modules'
+        key: npm-v3-${{ hashFiles('**/package-lock.json') }}
+    - name: Install dependencies
+      run: npm ci
+      shell: bash
+      if: steps.cache.outputs.cache-hit != 'true'
     - name: Install Foundry
       uses: foundry-rs/foundry-toolchain@v1
       with:
-        version: stable
\ No newline at end of file
+        version: nightly
```

Comment:
This might replace [some job steps in the tests](https://github.com/OpenZeppelin/uniswap-hooks/blob/master/.github/workflows/checks.yml#L10-L35). Could be for another PR